code/game/mecha/mecha.dm

```diff
@@ -584,10 +587,14 @@
 
 /obj/mecha/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum) //wrapper
 	log_message("Hit by [AM].")
+	if(isitem(AM))
```

Comment:
more than items can be thrown.

Reply:
Yeah I know. Thing is the whole throwing code is fucked up and throwingdatum is not passed.
I didn't want to go rework that entire system for now.
I did a quick analysis and a lot of stuff don't expect throwingdatum. A lot of ways in don't pass throwingdatum and so on. It's a mess